File: repos_estudiantes_ALVAREZ_primera_entrega/CARRERA_POZO_ALEJANDRA/Herencia/Ejercicio13/Empleado.py
class Empleado:
    def __init__(self, nombre, sueldomes):
        self.nombre = nombre
        self.sueldomes = sueldomes
    def sueldo_total(self):
        return self.sueldomes
    def __str__(self):
        return f"{self.__class__.__name__}: {self.nombre}, Sueldo mensual: {self.sueldomes}"
class Administrativo(Empleado):
    def __init__(self, nombre, sueldomes, cargo):
        super().__init__(nombre, sueldomes)
        self.cargo = cargo
class Chef(Empleado):
    def __init__(self, nombre, sueldomes, hora_extra, sueldo_hora, tipo):
        super().__init__(nombre, sueldomes)
        self.hora_extra = hora_extra
        self.sueldo_hora = sueldo_hora
        self.tipo = tipo
    def sueldo_total(self):
        return self.sueldomes + (self.hora_extra * self.sueldo_hora)
    
class Mesero(Empleado):
    def __init__(self, nombre, sueldomes, hora_extra, sueldo_hora, propina):
        super().__init__(nombre, sueldomes)
        self.hora_extra = hora_extra
        self.sueldo_hora = sueldo_hora
        self.propina = propina
    def sueldo_total(self):
        return self.sueldomes + (self.hora_extra * self.sueldo_hora) + self.propina

empleados = [
    Chef("Remy", 2500, 10, 15, "Francés"),
    Mesero("Alfredo", 1800, 5, 10, 120),
    Mesero("Colette", 1900, 6, 12, 150),
    Administrativo("Anton", 2200, "Contabilidad"),
    Administrativo("Linguini", 2000, "Recursos Humanos")]
X = 2000
print(f"\nEmpleados con sueldo mensual = {X}:")
for e in empleados:
    if e.sueldomes == X:
        print(f" - {e.nombre} ({e.__class__.__name__})")

print("\nSueldos totales:")
for e in empleados:
    print(f"{e.nombre}: {e.sueldo_total()}")

File: repos_estudiantes_TARQUI_primera_entrega/PRADO_FLORES_ARACELY_NAOMI/HERENCIA/HerenciaTrece/Empleado.java
package HerenciaTrece;

public class Empleado {
    protected String nombre;
    protected float sueldoMes;

    public Empleado() {
        this.nombre = "Desconocido";
        this.sueldoMes = 0;
    }

    public Empleado(String nombre, float sueldoMes) {
        this.nombre = nombre;
        this.sueldoMes = sueldoMes;
    }

    public void mostrar() {
        System.out.println("Nombre: " + nombre);
        System.out.println("Sueldo mensual: " + sueldoMes);
    }

    // Método polimórfico
    public float sueldoTotal() {
        return sueldoMes;
    }

    public float getSueldoMes() {
        return sueldoMes;
    }
}
